spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AlignMergeIntoTable.scala

```diff
@@ -22,12 +22,13 @@ package org.apache.spark.sql.catalyst.analysis
 import org.apache.spark.sql.AnalysisException
 import org.apache.spark.sql.catalyst.plans.logical.{Assignment, DeleteAction, InsertAction, LogicalPlan, MergeIntoTable, UpdateAction}
 import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.utils.AliasedIcebergTable
 import org.apache.spark.sql.internal.SQLConf
 
 case class AlignMergeIntoTable(conf: SQLConf) extends Rule[LogicalPlan] with AssignmentAlignmentSupport {
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-    case m: MergeIntoTable if m.resolved =>
+    case m@ MergeIntoTable(AliasedIcebergTable(_), _, _, _, _) if m.resolved =>
```

Comment:
I'm a little concerned about this change because it now depends on the number of arguments in `MergeIntoTable`. If Spark changes, then this rule would break.

Instead, could you move the inner match inside the rule? Like this:

```scala
  case m: MergeIntoTable if m.resolved =>
    m.relation match {
      case IcebergRelation(rel) =>
        ...
      case _ =>
        m
    }
```

Reply:
We could also expose a method called `isIcebergTable` or `isIcebergRelation` and append that to the `if` condition.

Reply:
That way, we keep the current indentation

Reply:
```
case m: MergeIntoTable if m.resolved && isIcebergRelation(m.relation) =>
  ....
```

Reply:
Yeah, I like Anton's suggestion better.

Reply:
@rdblue @aokolnychyi Since you both like to use `guards` .. i will switch to it. Thank you.